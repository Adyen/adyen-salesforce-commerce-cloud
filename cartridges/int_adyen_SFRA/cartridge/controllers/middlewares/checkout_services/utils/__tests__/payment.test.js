"use strict";

var handlePaymentAuthorization = require('../payment');

var _require = require('dw/order/OrderMgr'),
    getPaymentInstruments = _require.getPaymentInstruments;

var adyenHelpers = require('*/cartridge/scripts/checkout/adyenHelpers');

var res;
var emit;
var order;
beforeEach(function () {
  jest.clearAllMocks();
  res = {
    json: jest.fn()
  };
  emit = jest.fn();
  order = {
    getPaymentInstruments: getPaymentInstruments,
    orderNo: 'mocked_orderNo'
  };
});
describe('Payment', function () {
  it('should return json with error details when payment is unsuccessful', function () {
    adyenHelpers.handlePayments.mockReturnValue({
      error: true
    });
    var paymentCompleted = handlePaymentAuthorization({}, {
      res: res
    }, emit);
    expect(res.json.mock.calls).toMatchSnapshot();
    expect(paymentCompleted).toBeFalsy();
    expect(emit).toBeCalledWith('route:Complete');
  });
  it('should return 3ds2 json response', function () {
    adyenHelpers.handlePayments.mockReturnValue({
      threeDS2: true,
      resultCode: 'IdentifyShopper',
      action: 'mocked_action'
    });
    var paymentCompleted = handlePaymentAuthorization(order, {
      res: res
    }, emit);
    expect(res.json.mock.calls).toMatchSnapshot();
    expect(paymentCompleted).toBeFalsy();
    expect(emit).toBeCalledWith('route:Complete');
  });
  it('should return 3d json response', function () {
    adyenHelpers.handlePayments.mockReturnValue({
      redirectObject: {
        url: 'mocked_url',
        data: {
          PaReq: 'mocked_PaReq',
          MD: 'mocked_MD'
        }
      },
      signature: 'mocked_signature',
      authorized3d: true,
      orderNo: 'mocked_orderNo'
    });
    var paymentCompleted = handlePaymentAuthorization(order, {
      res: res
    }, emit);
    expect(res.json.mock.calls).toMatchSnapshot();
    expect(paymentCompleted).toBeFalsy();
    expect(emit).toBeCalledWith('route:Complete');
  });
  it('should redirect', function () {
    adyenHelpers.handlePayments.mockReturnValue({
      redirectObject: {
        url: 'mocked_url'
      },
      authorized3d: false,
      signature: 'mocked_signature',
      orderNo: 'mocked_orderNo'
    });
    var paymentCompleted = handlePaymentAuthorization(order, {
      res: res
    }, emit);
    expect(res.json.mock.calls).toMatchSnapshot();
    expect(paymentCompleted).toBeFalsy();
    expect(emit).toBeCalledWith('route:Complete');
  });
  it('should not redirect', function () {
    adyenHelpers.handlePayments.mockReturnValue({
      redirectObject: false,
      threeDS2: false
    });
    var paymentCompleted = handlePaymentAuthorization(order, {
      res: res
    }, emit);
    expect(paymentCompleted).toBeTruthy();
  });
});