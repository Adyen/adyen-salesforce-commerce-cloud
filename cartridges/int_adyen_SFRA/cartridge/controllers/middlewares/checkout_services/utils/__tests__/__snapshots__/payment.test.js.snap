// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment should redirect 1`] = `
Array [
  Array [
    Object {
      "continueUrl": "[\\"Adyen-Redirect\\",\\"merchantReference\\",\\"mocked_orderNo\\",\\"signature\\",\\"mocked_signature\\"]",
      "error": false,
    },
  ],
]
`;

exports[`Payment should return 3d json response 1`] = `
Array [
  Array [
    Object {
      "continueUrl": "[\\"Adyen-Adyen3D\\",\\"IssuerURL\\",\\"mocked_url\\",\\"PaRequest\\",\\"mocked_PaReq\\",\\"MD\\",\\"mocked_MD\\",\\"merchantReference\\",\\"mocked_orderNo\\",\\"signature\\",\\"mocked_signature\\"]",
      "error": false,
    },
  ],
]
`;

exports[`Payment should return 3ds2 json response 1`] = `
Array [
  Array [
    Object {
      "continueUrl": "[\\"Adyen-Adyen3DS2\\",\\"resultCode\\",\\"IdentifyShopper\\",\\"orderNo\\",\\"mocked_orderNo\\"]",
      "error": false,
      "order": Object {
        "getPaymentInstruments": [MockFunction] {
          "calls": Array [
            Array [
              "mocked_method",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": Object {
                "0": Object {
                  "custom": Object {
                    "adyenAction": "mocked_action",
                    "adyenMD": "mocked_adyen_MD",
                    "adyenPaymentData": "mocked_adyen_payment_data",
                    "adyenRedirectURL": "https://some_mocked_url/signature",
                  },
                },
                "iterator": [Function],
                "toArray": [MockFunction] {
                  "calls": Array [
                    Array [],
                  ],
                  "results": Array [
                    Object {
                      "type": "return",
                      "value": Array [
                        Object {
                          "custom": Object {
                            "adyenAction": "mocked_action",
                            "adyenMD": "mocked_adyen_MD",
                            "adyenPaymentData": "mocked_adyen_payment_data",
                            "adyenRedirectURL": "https://some_mocked_url/signature",
                          },
                        },
                      ],
                    },
                  ],
                },
              },
            },
          ],
        },
        "orderNo": "mocked_orderNo",
      },
    },
  ],
]
`;

exports[`Payment should return json with error details when payment is unsuccessful 1`] = `
Array [
  Array [
    Object {
      "error": true,
      "errorMessage": "mocked_error.payment.not.valid",
    },
  ],
]
`;
